llvm/lib/Object/SymbolSize.cpp

```diff
@@ -59,6 +59,13 @@ llvm::object::computeSymbolSizes(const ObjectFile &O) {
     return Ret;
   }
 
+  if (const auto *E = dyn_cast<XCOFFObjectFile>(&O)) {
+    auto Syms = E->symbols();
+    for (XCOFFSymbolRef Sym : Syms)
```

Comment:
since in class ObjectFile , we already have 

```
  using symbol_iterator_range = iterator_range<symbol_iterator>;
  symbol_iterator_range symbols() const {
    return symbol_iterator_range(symbol_begin(), symbol_end());
  } 
``` 

I do not think you need to define  a new class xcoff_symbol_iterator , 
 we can change 
 change 
` for (XCOFFSymbolRef Sym : Syms) `
 --->
` for (SymbolRef Sym : Syms) `
 
 and cast SymbolRef  to XCOFFSymbolRef here.

Reply:
> I do not think you need to define a new class xcoff_symbol_iterator ,

To me, adding a iterator range seems natural when there are already `symbol_begin` and `symbol_end` inside a class? (Maybe we can change `symbol_begin` and `symbol_end` with the new `xcoff_symbol_iterator` class). What do you think?